signature/jws/signer.go

```diff
@@ -68,31 +55,99 @@ func NewSignerWithCertificateChain(method jwt.SigningMethod, key crypto.PrivateK
 		return nil, err
 	}
 
+	keySpec, err := keySpecFromJWS(method.Alg())
+	if err != nil {
+		return nil, err
+	}
+
 	rawCerts := make([][]byte, len(certChain))
 	for i, cert := range certChain {
 		rawCerts[i] = cert.Raw
 	}
 	return &Signer{
-		method:    method,
-		key:       key,
-		certChain: rawCerts,
+		runner: &builtinPlugin{
+			keySpec:   keySpec,
+			method:    method,
+			key:       key,
+			certChain: rawCerts,
+		},
 	}, nil
 }
 
-// Sign signs the artifact described by its descriptor, and returns the signature.
-func (s *Signer) Sign(ctx context.Context, desc notation.Descriptor, opts notation.SignOptions) ([]byte, error) {
-	// generate JWT
-	payload := packPayload(desc, opts)
-	if err := payload.Valid(); err != nil {
-		return nil, err
+// builtinPlugin is a plugin.Runner implementation which
+// signs supports the generate-signature workflow using
+// the provided key and certificates.
+type builtinPlugin struct {
+	keySpec notation.KeySpec
+
+	// method is the method to sign artifacts.
+	method jwt.SigningMethod
+
+	// key is the signing key used to sign artifacts.
+	// The signing key can be either remote or local.
+	key crypto.PrivateKey
```

Comment:
Not sure why we say this can be a remote or local key, it would always be a local key on disk.

Reply:
Done